app/controllers/curation_concerns/generic_works_controller.rb

```diff
@@ -16,6 +16,16 @@ def destroy
     end
     report_status_and_redirect(asset_with_relationships, batch: false)
   end
+  # we have to override this action in Curation Concerns application controller behavior class
+
+  def deny_access
+    locals = UnauthorizedPresenter.new(params).render_asset_401
+
+    render template: '/error/unauthorized',
+           formats: [:html],
+           status: 401,
+           locals: { title: locals[:title], message: locals[:message], depositor_first_name: locals[:depositor_first_name], depositor_last_name: locals[:depositor_last_name], thumbnail: locals[:thumbnail] }
```

Comment:
why all the local variables? The only thing you need to pass to the view is the presenter. The presenter should have methods for title, message, depositor, etc.